src/box/port.c

```diff
@@ -37,34 +37,101 @@
 #include <fiber.h>
 #include "errinj.h"
 #include "mpstream/mpstream.h"
+#include "mp_tuple.h"
 #include "tweaks.h"
 #include "box/mp_box_ctx.h"
 
 /**
  * The pool is used by port_c to allocate entries and to store
- * result data when it fits into an object from the pool.
+ * result data when it fits into an object from the pool. Also,
+ * it is used to store `struct port_c_iterator`.
  */
 static struct mempool port_entry_pool;
 
+/**
+ * The pool is used by port_c to allocate mp_ctx objects because
+ * they are too large to store them right in entries.
+ */
+static struct mempool port_mp_ctx_pool;
+
 enum {
 	PORT_ENTRY_SIZE = sizeof(struct port_c_entry),
+	PORT_MP_CTX_SIZE = sizeof(struct mp_ctx),
```

Comment:
Let's store `mp_ctx` in `port_c_entry`?

Reply:
`mp_ctx` is huge - four times larger than `port_c_entry`.